libnavigation-core/src/main/java/com/mapbox/navigation/core/lifecycle/ForwardMapboxNavigation.kt

```diff
@@ -0,0 +1,29 @@
+@file:JvmName("ForwardMapboxNavigation")
+
+package com.mapbox.navigation.core.lifecycle
+
+import com.mapbox.navigation.core.MapboxNavigation
+
+/**
+ * This extension removes boilerplate from a class that needs to use [MapboxNavigation], and it
+ * makes your class appear as if it implements [MapboxNavigationObserver] without exposing the
+ * functions.
+ *
+ * ``` kotlin
+ * val navigationObserver = mapboxNavigationForward(this::onAttached, this::onDetached)
+ * private fun onAttached(mapboxNavigation: MapboxNavigation)
+ * private fun onDetached(mapboxNavigation: MapboxNavigation)
+ * ```
+ */
+fun forwardMapboxNavigation(
```

Comment:
It's weird that we now have `mapboxNavigationForward` and `forwardMapboxNavigation` with identical implementations. cc @kmadsen

Reply:
Yeah, I had some direct messages with `abhishek1508` about this. I really like the `mapboxNavigationForward` function in android auto. It is a little similar to `requireMapboxNavigation`

So the plan will be to delete the `mapboxNavigationForward` function in androidauto, in favor of `forwardMapboxNavigation` in the core-sdk. Sorry it's confusing to have them renamed, consequence of iteration.

But this is all intentional